devlib/module/thermal.py

```diff
@@ -105,3 +125,10 @@ def disable_all_zones(self):
         """Disables all the thermal zones in the target"""
         for zone in self.zones.values():
             zone.set_enabled(False)
+
+    def get_all_temperatures(self):
```

Comment:
Given our offline discussion on error handling, you might want to add an `error='raise'` parameter, that the user could set to `raise` or `ignore`. `ignore` will just ignore the error and not add the key in the returned dict, `raise` would re-raise the exception.

Reply:
The `ignore` case should probably log at error level still if the reason of the problem is a driver issue (i.e. not part of normal operation for that file), so consumers have a chance of knowing there is a problem related to that if some other code misbehaves (e.g. code iterating on zones then looking up in the dict and failing with a `KeyError`)